app-commons/src/main/scala/org/bitcoins/commons/jsonmodels/dlc/DLCStatus.scala

```diff
@@ -0,0 +1,295 @@
+package org.bitcoins.commons.jsonmodels.dlc
+
+import org.bitcoins.commons.jsonmodels.dlc.DLCMessage.{
+  DLCAccept,
+  DLCMutualCloseSig,
+  DLCOffer,
+  DLCSign
+}
+import org.bitcoins.core.protocol.transaction.Transaction
+import org.bitcoins.crypto.{SchnorrDigitalSignature, Sha256DigestBE}
+
+sealed trait DLCStatus {
+  def eventId: Sha256DigestBE
+  def isInitiator: Boolean
+  def offer: DLCOffer
+}
+
+sealed trait AcceptedDLCStatus extends DLCStatus {
+  def accept: DLCAccept
+}
+
+object DLCStatus {
+
+  def statusString(status: DLCStatus): String = {
+    status match {
+      case _: Offered         => "OFFERED"
+      case _: Accepted        => "ACCEPTED"
+      case _: Signed          => "SIGNED"
+      case _: Broadcasted     => "BROADCASTED"
+      case _: Confirmed       => "CONFIRMED"
+      case _: CloseOffered    => "CLOSE OFFERED"
+      case _: Closed          => "CLOSED"
+      case _: Claiming        => "CLAIMING"
+      case _: Claimed         => "CLAIMED"
+      case _: Penalized       => "PENALIZED"
+      case _: RemoteClaiming  => "REMOTE CLAIMING"
+      case _: RemoteClaimed   => "REMOTE CLAIMED"
+      case _: RemotePenalized => "REMOTE PENALIZED"
+      case _: Refunded        => "REFUNDED"
+    }
+  }
+
+  case class Offered(
+      eventId: Sha256DigestBE,
+      isInitiator: Boolean,
+      offer: DLCOffer)
+      extends DLCStatus {
+
+    def toAccepted(accept: DLCAccept): Accepted = {
+      Accepted(eventId, isInitiator, offer, accept)
+    }
+  }
+
+  case class Accepted(
+      eventId: Sha256DigestBE,
+      isInitiator: Boolean,
+      offer: DLCOffer,
+      accept: DLCAccept)
+      extends AcceptedDLCStatus {
+
+    def toSigned(sign: DLCSign): Signed = {
+      Signed(eventId, isInitiator, offer, accept, sign)
+    }
+  }
+
+  case class Signed(
```

Comment:
So at this state either party can broadcast the DLC and have it confirmed on the chain right? It's probably a good idea to have this state, but we need to be careful when handling it as I think it is effectively equivalent to confirmed. 

Bitcoin exchanges use this semantic whenever they sign transactions internally because as soon as you authorize a transaction with digital signatures, it can be broadcast by anyone and have funds sent from your wallet. Until this is double spent, or entirely purged from your system (very hard to do!), these funds are at risk of leaving your wallet. 

If you issue multiple `Signed` to your counter party, and they don't use the same utxos, they can broadcast both of them and steal money from you. I guess there isn't an actionable item here, but more of a warning. 

Perhaps it's worth creating a higher level type in the ADT representing when funds are sign by both parties and are in flight. Is that what `AcceptedDLCStatus` is?